internal/store/atlas/clusters.go

```diff
@@ -0,0 +1,39 @@
+// Copyright 2020 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package atlas
+
+import (
+	"fmt"
+
+	"github.com/mongodb/mongodb-atlas-cli/internal/config"
+	atlas "go.mongodb.org/atlas/mongodbatlas"
+)
+
+//go:generate mockgen -destination=../mocks/mock_clusters.go -package=mocks github.com/mongodb/mongodb-atlas-cli/internal/store ClusterLister,AtlasClusterDescriber,OpsManagerClusterDescriber,ClusterCreator,ClusterDeleter,ClusterUpdater,AtlasClusterGetterUpdater,ClusterPauser,ClusterStarter,AtlasClusterQuickStarter,SampleDataAdder,SampleDataStatusDescriber,AtlasClusterConfigurationOptionsDescriber,AtlasSharedClusterDescriber,ClusterUpgrader,AtlasSharedClusterGetterUpgrader,AtlasClusterConfigurationOptionsUpdater,ClusterTester
+
+type ClusterLister interface {
+	ProjectClusters(string, *atlas.ListOptions) (interface{}, error)
+}
+
+// ProjectClusters encapsulate the logic to manage different cloud providers.
+func (s *Store) ProjectClusters(projectID string, opts *atlas.ListOptions) (interface{}, error) {
+	switch s.service {
```

Comment:
[nit] I would argue if we need switch case. All other stores in atlas do not differenciate between those as we do not need it.

Reply:
True, this store still needs to be migrated in a separate PR, so may go ahead and merge this one and suggest that change in the migration follow-up

Reply:
@matt-condon Feel free to leave it as we might apply some regexp to remove the switch case globally.
Sadly copilot doesn't help with this case.